### DISCUSSION TITLE
Prisma Schema Model - No negative numbers

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#8923] [Q&A]
Hello, 

I would like to define a field in my model which represents the amount of a euro withdrawal. 
Its constraints are : 
- must be positive or null (otherwise it would mean to add some money)
- must include up to two decimals. 
- avoid any rounded mistakes while doing sum or substraction operation

I ended up by writing this : 

>  valueEur     Decimal  @db.Decimal(10, 2)

But this does not prevent any negative number to be inserted. 
Is there any better solution ?

Thank you very much in advance :)

**Comment by ryands17** [comment]
@laurent512 👋 

You can add a [check constraint](https://www.postgresqltutorial.com/postgresql-check-constraint/) in your migration `.sql` file.

1. Create a migration using `prisma migrate dev --create-only`
2. Add the check constraint to the field in the created `.sql` migration file
3. Apply using `prisma migrate dev`

This should prevent -ve numbers being inserted.

**Reply by laurent512** [reply]
Sorry for the very dumb question, but is there a way to bypass the use of migrate dev ? 
I am hosted on heroku, cannot create table and shadow table are not working (just tried)

Would there be a way to solve that by generating the .sql file and then pushing directly?

**Reply by ryands17** [reply]
That's not possible. You need to create the shadow database as shown [here](https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database#cloud-hosted-shadow-databases-must-be-created-manually).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
